src/github.com/matrix-org/dendrite/cmd/dendrite-roomserver/main.go

```diff
@@ -43,10 +45,13 @@ func main() {
 	}
 
 	consumer := input.Consumer{
-		Consumer:             kafkaConsumer,
+		ContinualConsumer: common.ContinualConsumer{
```

Comment:
It feels a bit weird for half the fields of ContinualConsumer to be populated here and the others be populated by the Consumer when it is started. It feels to me like the Consumer should be responsible for making a new `ContinualConsumer` and copying the fields in.

YMMV though, certainly happy to go with this.